src/api/resetIndex.js

```diff
@@ -26,38 +28,57 @@ import { resolveFilepath } from '../utils/resolveFilepath.js'
  * await git.resetIndex({ fs, dir: '/tutorial', filepath: 'README.md' })
  * console.log('done')
  *
+ * @example
+ * await git.resetIndex({ fs, dir: '/tutorial', filepath: 'README.md', oid: '11a57ba27f571e9e2bfa344fdbbccad1d0e34463' })
+ * console.log('done')
+ *
  */
 export async function resetIndex({
   fs: _fs,
   dir,
   gitdir = join(dir, '.git'),
   filepath,
-  ref = 'HEAD',
+  ref,
+  oid,
   cache = {},
 }) {
   try {
     assertParameter('fs', _fs)
     assertParameter('gitdir', gitdir)
     assertParameter('filepath', filepath)
-    assertParameter('ref', ref)
 
     const fs = new FileSystem(_fs)
-    // Resolve commit
-    let oid = await GitRefManager.resolve({ fs, gitdir, ref })
+
     let workdirOid
-    try {
-      // Resolve blob
-      oid = await resolveFilepath({
-        fs,
-        cache,
-        gitdir,
-        oid,
-        filepath,
-      })
-    } catch (e) {
-      // This means we're resetting the file to a "deleted" state
-      oid = null
+
+    if (!oid) {
+      try {
+        // Resolve ref if no oid is provided
+        oid = await GitRefManager.resolve({ fs, gitdir, ref: ref || 'HEAD' })
+      } catch (e) {
+        if (ref) {
```

Comment:
If a `ref` is explicitly provided, we rethrow the error indicating the ref could not be resolved. If no `ref` is provided, we just continue.

This is in analogy to how `git reset` works on an empty repository. If you run `git reset`, it will remove all the files from the index. If you run `git reset HEAD`, it will throw an error.

This is the reason I had to remove the default value for `ref` in the argument list.

Reply:
I would not remove the default `'HEAD'` from the parameter, this is almost the same as `ref || 'HEAD'`. But the default value in the function signature is some kind of documentation.